codersdk/workspacebuilds.go

```diff
@@ -49,8 +49,9 @@ type WorkspaceBuild struct {
 	InitiatorID        uuid.UUID           `json:"initiator_id"`
 	InitiatorUsername  string              `json:"initiator_name"`
 	Job                ProvisionerJob      `json:"job"`
-	Deadline           NullTime            `json:"deadline,omitempty"`
 	Reason             BuildReason         `db:"reason" json:"reason"`
+	Deadline           NullTime            `json:"deadline,omitempty"`
+	Resources          []WorkspaceResource `json:"resources,omitempty"`
```

Comment:
If this isn't always returned, it shouldn't be a field on this object. I think our API should be fully representative of the data it returns, not extra fields for some endpoints.

I see this is used by the SSE... I'd propose that we make that an event-based system rather than a database polling one using the Pubsub.

Reply:
I talked with a few people and settled on DB polling for now and pubsub in a separate PR to keep this small. The main goal here is as much to lay the ground work for SSE and also cut down on round trips. My back of the napkin math says this will not result in more queries per second per dashboard client to the DB since we are skipping multiple middleware now. 

Let's chat about the resources stuff because I want to know more what you mean.

Reply:
If I returned this data on the other endpoints to unify the behavior and returned `[]` when there are no related resources, would that satisfy your issues? I just know it's way nicer for the client to consume this data as a whole.